mobile/src/components/courses/courses.tsx

```diff
@@ -49,26 +57,58 @@ const Courses: FC = (): ReactElement => {
   };
 
   const handleSearch = (search: string): void => {
-    dispatch(coursesActions.getCourses({ title: search, categoryKey: '' }));
+    if (!courseCategory) {
+      dispatch(coursesActions.getCourses({ title: search, categoryKey: '' }));
+    } else {
+      dispatch(
+        coursesActions.getCourses({
+          title: search,
+          categoryKey: courseCategory.key,
+        }),
+      );
+    }
```

Comment:
It's better to modify payload outside call, and then make it only in one place
```suggestion
    dispatch(coursesActions.getCourses(payload));
```